spec/requests/adminpanel_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe '/adminpanel', type: :request do
+  before(:each) do
+    user = User.create!(email: 'test@example.com', username: 'test', password: 'f4k3p455w0rd', user_type: 'administrator')
+    user.confirm
+    login_as(user, scope: :user)
+  end
+
+  describe 'GET /index' do
+    it 'renders a successful response' do
+      users = User.create(email: 'example@example.com', username: 'example', password: 'dhsjhgd67', user_type: 'administrator')
+      get adminpanel_url(users)
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'GET /edit' do
+    it 'render a successful response' do
+      users = User.create!(email: 'example@example.com', username: 'example', password: 'dhsjhgd67', user_type: 'administrator')
+      get edit_adminpanel_path(users)
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'PATCH /update' do
+    context 'with valid parameters' do
+      let(:new_attributes) {
+        { user_id: '7', username: 'example1', user_type: 'user' }
```

Comment:
Just as a suggestion, remember that tests must always be passed, in this case it would assume that when running a seed, you have a user with id = 7, but if not, the test may fail.

Reply:
Yes thank you very much, I took your suggestions and fixed the code.